include/libudiv/udiv.h

```diff
@@ -0,0 +1,124 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+
+   KallistiOS ##version##
+
+   include/libudiv/udiv.h
+   Copyright (C) 2024 Paul Cercueil <paul@crapouillou.net>
+*/
+
+/** \file    libudiv/udiv.h
+    \brief   Fast uint32_t division routines
+    \ingroup libudiv
+
+    This file was originally from `libudiv`.
+
+    \author Paul Cercueil
+*/
+
+#ifndef __LIBUDIV_H__
+#define __LIBUDIV_H__
+
+#include <kos/cdefs.h>
+
+/** \defgroup   libudiv     libudiv
+    \brief      Optimized unsigned 32-bit division routines
+    \ingroup    math_general
+
+    libudiv is an optimized math library that provides routines for efficiently
+    dividing an unsigned 32-bit variable by a known unsigned 32-bit constant.
```

Comment:
libudiv is a small library that provides optimized unsigned 32-bit division routines, to be used when the divisor is constant or seldomly updated.